pkg/devfile/adapters/docker/component/utils.go

```diff
@@ -125,27 +134,73 @@ func (a Adapter) pullAndStartContainer(componentName string, projectVolumeName s
 	s.End(true)
 
 	// Start the container
-	err = a.startContainer(componentName, projectVolumeName, comp)
+	err = a.startContainer(componentName, projectVolumeName, supervisordVolumeName, comp)
 	if err != nil {
 		return errors.Wrapf(err, "Unable to start container for devfile component %s", *comp.Alias)
 	}
 	glog.V(3).Infof("Successfully created container %s for component %s", *comp.Image, componentName)
 	return nil
 }
 
-func (a Adapter) startContainer(componentName string, projectVolumeName string, comp versionsCommon.DevfileComponent) error {
-	containerConfig := a.generateAndGetContainerConfig(componentName, comp)
+func (a Adapter) startContainer(componentName string, projectVolumeName, supervisordVolumeName string, comp versionsCommon.DevfileComponent) error {
 	hostConfig := container.HostConfig{}
 
+	runCommand, err := adaptersCommon.GetRunCommand(a.Devfile.Data, a.devfileRunCmd)
```

Comment:
Can we move the runCommand logic outside of the `startContainer` function since this function will be used to start containers other than just the container that the run command is pointing to?

Plus I'm concerned this would cause issues for any non "run-command" containers started.

Reply:
Not sure where else this logic would be moved out to, if this is done outside `startContainer`/`pullAndStartContainer`, I would have to call a function that does this check everytime a container is started, as opposed to this PR, which does it only once in the `startContainer`

I have made sure that this does interfere with non run-commands since `startContainer` loops thru the devfile components and this logic just checks if runCommand actions references any of the devfile components... 

What do you think?

Reply:
Hmm, fair point.

What about:
- Renaming the function to something other than `startContainer` (startComponent?)
- Moving the supervisord volume mounting to its own function (that `startContainer`/`startComponent` calls)? 

This way it keeps things a little cleaner and easier to unit test

Reply:
already done before you commented lol